add Japanese translation

nyampire opened the pull request:
Hello team,
I've translated Streetmix into Japanese. (except "text" in segment-info.json)
I appreciate if this is work for you.

treyhahn commented:
Hey @nyampire, this is great! Thanks! How did you hear about Streetmix? Have you used it yourself before?

nyampire commented on assets/locales/jp/main.json:
```diff
@@ -0,0 +1,289 @@
+      "about": "Streetmixについて",
+      "contact": "コンタクト",
+      "community": "コミュニティ",
+      "contribute": "Contribute",
```
Traditionally, "貢献" is the Japanese word for Open Source contribution. But this word sounds slightly bookish style. So, I have 2 choices. I prefer 2nd.
1. Use "貢献" for a while.
2. Keep it in English for a while, and seek more "soft" wording.

nyampire commented on assets/locales/jp/main.json:
```diff
@@ -0,0 +1,289 @@
+      "option-sky": "空を透明化",
+      "loading": "ロード中…",
+      "save-button": "コンピュータに保存中…",
+      "license": "Streetmixで作成した画像は <br><a href='{url}'>Creative Commons Attribution-ShareAlike 4.0 International License</a> のもとで自由にお使いいただけます。",
```
Sorry I forgot to work this line.
It is "クリエイティブ・コモンズ 表示 - 継承 4.0 国際". URL is [here](https://creativecommons.org/licenses/by-sa/4.0/deed.ja).

nyampire commented on assets/locales/jp/main.json:
```diff
@@ -0,0 +1,289 @@
+    "seconds-ago": "数秒以内",
+    "yesterday": "昨日 {time}",
+    "today": "本日 {time}",
+    "format-date-no-year": "MMM D",
```
Abbreviation is not common in Japanese. I prefer "mm-dd", "mm/dd", or "mm月 dd日". 
I think "mm月 dd日" is better wording for this place.

nyampire commented on assets/locales/jp/main.json:
```diff
@@ -0,0 +1,289 @@
+  "meta": {
+    "description": "あなたの身近な道路をデザインし、リミックスし、シェアしましょう。必要なものはブラウザだけです！ 街路樹や自転車レーン、歩道や車道を追加してみて、あなたのアイデアで地域コミュニティがどう変わるのかを試してみましょう。"
+  },
+  "i18n": {
```
"Japanese (ja)" is better. Could you add this line and fix the folder name?

nyampire commented:
Thank you for reviewing @treyhahn ! I'm pretty happy to contribute this lovely app.
As I'm a contributor for OpenStreetMap and CivicTech, I had heard of Streetmix few yeas ago.
And some week ago, @mapmeld has introduced this app to me. I feel that it is nicer time to contribute :)

treyhahn commented:
Great, thanks! I don't speak Japanese and I don't want to potentially mess up the translation, so can you make your changes to `クリエイティブ・コモンズ 表示 - 継承 4.0 国際` and `mm月 dd日` in another commit to this branch?

treyhahn commented:
I'm glad you found us! What are your thoughts on the app?

treyhahn commented:
@nyampire Also, is it easy to translate the languages into Japanese? E.g. `日本語` instead of `Japanese` (I don't know if that's correct, i just copied it from Google Translate), and do that for the rest of the languages listed as well?

nyampire commented:
Thanks. I've updated the lines.

https://github.com/streetmix/streetmix/pull/903#issuecomment-379421638
Translation itself is easy. But I think those lines are for non-Japanese speakers. So it is better to keep them in English.
e.g. If the text `英語` (English) is shown on the page, non-Japanese speaker cannot understand what it is. So they could not switch the languages.

treyhahn commented:
Thanks for the feedback about switching languages! I think that's a really good point to bring up. We've heard that from another person before, and now I've noted your feedback too.

Did you decide to switch to the other option for `contribute`? That's totally fine if you think it's best, I'm just double checking for documentation.

Just in case I missed something I'm going to ask @louh to review this. He's on vacation for a few days, so this will probably get merged when he gets back.

louh commented:
Thank you so much for the work and progress here, @nyampire and @treyhahn.

We are currently using translated locale names in the language selection. It is true that this makes it unreadable for speakers of another language. I've seen it done both ways before. We can switch the display to using the locale's own specific name, but this can be done easily through code. We cannot do it the other way around without the translation. I would suggest translating all the locale names anyway, which gives us maximum flexibility if we change the behavior of the language dropdown later.

nyampire commented:
Thank you, @louh and @treyhahn . I see.
I've added Japanese translation about language names. (although "en(pirate)" is deleted in #904, but I added it just in case of.)

And I decided to use traditional word `貢献` to `contribution`. (also translated already)
Because of consistency for other lines and other Open Source Projects.
